src/wallet/KintoWallet.sol

```diff
@@ -212,7 +213,7 @@ contract KintoWallet is Initializable, BaseAccount, TokenCallbackHandler, IKinto
      */
     function completeRecovery(address[] calldata newSigners) external override onlyFactory {
         require(inRecovery > 0 && block.timestamp > (inRecovery + RECOVERY_TIME), "KW-fr: too early");
-        require(!kintoID.isKYC(owners[0]), "KW-fr: Old KYC must be burned");
+        require(!kintoID.isKYC(owners[0]) && kintoID.isKYC(newSigners[0]), "KW-fr: Old KYC must have been transferred");
```

Comment:
I believe this check is already performed by `transferOnRecovery`, right?
```
require(balanceOf(_from) > 0 && balanceOf(_to) == 0, "Invalid transfer");
```

Reply:
but thats on kintoid, we dont know necessarily that those are the owners

Reply:
what i'm saying is that I think we could simply have this `completeRecovery` this way:

```
function completeRecovery(address[] calldata newSigners) external override onlyFactory {
        require(inRecovery > 0 && block.timestamp > (inRecovery + RECOVERY_TIME), "KW-fr: too early");
        require(msg.sender == IKintoWallet(wallet).recoverer(), "only recoverer");
        // Transfer kinto id from old to new signer
        kintoID.transferOnRecovery(IKintoWallet(wallet).owners(0), newSigners[0]);

        _resetSigners(newSigners, SINGLE_SIGNER);
        inRecovery = 0;
    }
    ```
    So, the flow is:
    1. walletFactory.completeWalletRecovery() (only checks that the wallet exists in the factory and that the sender is the recoverer). And, actually, we could even send the sender to the `completeRecovery` and check that condition on the `completeRecovery`
    2. the factory calls `completeRecovery()`:
    - checks `inRecovery` and time elapsed
    - calls `transferOnRecovery`
    
    3. the call to `kintoID.transferOnRecovery` performs a check that the sender (`owner(0)`) has a KYC and that the receiver ( `newSigners[0]`) does not have any so it can do the transfer